src/main/java/wooteco/subway/domain/section/SectionDeleteStatus.java

```diff
@@ -0,0 +1,54 @@
+package wooteco.subway.domain.section;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.function.BiPredicate;
+import java.util.stream.Stream;
+
+public enum SectionDeleteStatus {
+    DELETE_UP_STATION(SectionDeleteStatus::isFirstSection),
+    DELETE_DOWN_STATION(SectionDeleteStatus::isLastSection),
+    DELETE_MIDDLE(SectionDeleteStatus::isMiddleSection),
+    ;
+
+    private final BiPredicate<List<Section>, Long> condition;
+
+    SectionDeleteStatus(final BiPredicate<List<Section>, Long> condition) {
+        this.condition = condition;
+    }
+
+    public static SectionDeleteStatus from(final List<Section> sections, final Long stationId) {
+        return Arrays.stream(values())
+            .filter(it -> it.condition.test(sections, stationId))
+            .findFirst()
+            .orElseThrow(() -> new IllegalStateException("[ERROR] 해당 구간을 삭제할 수 없습니다."));
+    }
+
+    @SuppressWarnings("OptionalGetWithoutIsPresent")
+    private static boolean isFirstSection(final List<Section> sections, final Long stationId) {
+        final Long upStationId = sections.stream()
+            .flatMap(section -> Stream.of(section.getUpStationId(), section.getDownStationId()))
+            .distinct()
+            .min(Comparator.comparing(it -> it))
+            .get();
+        return upStationId.equals(stationId);
+    }
+
+    @SuppressWarnings("OptionalGetWithoutIsPresent")
+    private static boolean isLastSection(final List<Section> sections, final Long stationId) {
+        final Long downStationId = sections.stream()
+            .flatMap(section -> Stream.of(section.getUpStationId(), section.getDownStationId()))
+            .distinct()
+            .max(Comparator.comparing(it -> it))
+            .get();
+        return downStationId.equals(stationId);
+    }
+
+    private static boolean isMiddleSection(final List<Section> sections, final Long stationId) {
+        if (isFirstSection(sections, stationId)) {
+            return false;
+        }
+        return !isLastSection(sections, stationId);
```

Comment:
```suggestion
        return !isFirstSection(sections, stationId) && !isLastSection(sections, stationId);
```
이 부분은 취향차일 수 있으니 이렇게도 쓸 수 있다 정도만 알면 좋을 것 같네요. 이미 아실 수도 있구요 😄

Reply:
after 상태의 코드를 before 코드로 수정한 내역이었습니다!

최근에 boolean식과 드모르간 법칙에 대해 학습하고 적용하려 애쓰는 중이었습니다!
빨리 걸러지는 로직을 early return하면 효율상 이점을 가질 수 있고,  부정문도 줄어든다는 장점이 있었습니다!

범블비 생각은 어떠신지요?!

https://blog.chojaeseong.com/java/if/conditionalstatements/a&&b/earlyreturn/boolean/%EC%9A%B0%ED%85%8C%EC%BD%94/2022/03/30/%EB%B6%84%EA%B8%B0%EB%AC%B8%EA%B3%BC-%EB%93%9C%EB%AA%A8%EB%A5%B4%EA%B0%84.html

Reply:
저도 최근에 이런 스타일에 대해서 알게되었는데 깔끔하게 글로 정리해주셨네요. 감사합니다 🙇